### PR TITLE
check if we run in a container

### PR DESCRIPTION
**Pull request by timar**
Signed-off-by: Andras Timar <andras.timar@collabora.com>
Change-Id: Idc9db40c00d41c95160db130eb324c487f14de17

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: common/security.h [hunk 1]
```diff
@@ -32,6 +32,30 @@ inline int hasUID(const char *userId)
...
+    cgroup = fopen("/proc/self/cgroup", "r");
+    if(!cgroup)
+    {
+        fprintf(stderr, "Error: cannot open /proc/self/cgroup\n");
+        return 0;
+    }
+    if (fgets(line, sizeof(line), cgroup) == NULL)
+    {
+        fprintf(stderr, "Error: cannot read from /proc/self/cgroup\n");
+        return 0;
```
Comment (kendy) [inline common/security.h:50]: Shouldn't we fclose(cgroup) here too?

File: common/security.h [hunk 2]
```diff
@@ -32,6 +32,30 @@ inline int hasUID(const char *userId)
...
+    }
+    fclose(cgroup);
```
Comment (kendy) [inline common/security.h:52]: I guess we should line[sizeof(line)-1] = 0; here before the strstr to be safe?

File: common/security.h (hunk 1 above)
Comment (kendy) [reply common/security.h:50]: I mean, one line up, before the return obviously :-)

File: common/security.h (hunk 2 above)
Comment (timar) [reply common/security.h:55]: fgets() inserts the terminating \0

File: common/security.h (hunk 1 above)
Comment (timar) [reply common/security.h]: I fixed this in latest version of PR.

**Review by mmeeks — APPROVED** [review]
Looks great to me =) thanks for the feedback etc. - good to have lots of eyes on this sort of thing.

File: common/security.h (hunk 2 above)
Comment (kendy) [reply common/security.h:55]: Ah you are right, thanks :-)

**Pull request merged**